File: test/fixtures/ast-grep/no-weak-hash-ts/ok.ts
import crypto, { createHash, createHmac } from 'crypto';

// OK: SHA-256 is a current, acceptable digest
export function sha256(value: string): string {
  return crypto.createHash('sha256').update(value).digest('hex');
}

// OK: near-miss - sha512 contains neither "md5" nor a bare "sha1"
export function sha512(value: string): string {
  return createHash('sha512').update(value).digest('hex');
}

// OK: near-miss - the algorithm name appears only as data, not as a call
export const legacyAlgorithms = ['md5', 'sha1'];

// OK: near-miss - HMAC with a modern digest
export function sign(value: string, key: string): string {
  return createHmac('sha256', key).update(value).digest('hex');
}

File: test/fixtures/ast-grep/no-weak-hash-java/ok.java
package fixtures;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.security.Signature;
import javax.crypto.Cipher;
import javax.crypto.Mac;
import javax.crypto.NoSuchPaddingException;

public class Ok {

    // OK: SHA-256 is a current, acceptable digest
    public MessageDigest sha256() throws NoSuchAlgorithmException {
        return MessageDigest.getInstance("SHA-256");
    }

    // OK: near-miss - SHA-512 must not be caught by a sloppy "sha" prefix match
    public MessageDigest sha512() throws NoSuchAlgorithmException {
        return MessageDigest.getInstance("SHA-512");
    }

    // OK: near-miss - the algorithm names appear only as data
    public static final String[] LEGACY_ALGORITHMS = { "MD5", "SHA-1" };

    // OK: near-miss - a modern HMAC
    public Mac hmac() throws NoSuchAlgorithmException {
        return Mac.getInstance("HmacSHA256");
    }

    // OK: near-miss - a SHA-256 signature
    public Signature signature() throws NoSuchAlgorithmException {
        return Signature.getInstance("SHA256withRSA");
    }

    // OK: near-miss - authenticated AES-GCM, not DES and not ECB
    public Cipher cipher() throws NoSuchAlgorithmException, NoSuchPaddingException {
        return Cipher.getInstance("AES/GCM/NoPadding");
    }
}
